src/main/java/org/apache/sysml/runtime/instructions/gpu/context/GPUObject.java

```diff
@@ -858,7 +827,7 @@ void copyFromHostToDevice(String opcode) throws DMLRuntimeException {
 
 			if (copyToDevice) {
 				long t1 = GPUStatistics.DISPLAY_STATISTICS ? System.nanoTime() : 0;
-				CSRPointer.copyToDevice(getJcudaSparseMatrixPtr(), tmp.getNumRows(), tmp.getNonZeros(), rowPtr, colInd,
+				CSRPointer.copyToDevice(gCtx, getJcudaSparseMatrixPtr(), tmp.getNumRows(), tmp.getNonZeros(), rowPtr, colInd,
```

Comment:
in the line `CSRPointer.copyToDevice`, please use `getGPUContext()` instead of the passed in `GPUContext gCtx` parameter. Please get rid of that parameter.

Reply:
Done